sql/src/test/java/io/crate/planner/operators/PushDownTest.java

```diff
@@ -359,6 +359,25 @@ public void testWhereClauseIsPushedDownIntoSubRelationOfUnion() {
         );
     }
 
+    @Test
+    public void test_filter_is_pushed_beneath_project_set_if_it_targets_standalone_outputs() {
+        var plan = sqlExecutor.logicalPlan(
+            "SELECT" +
+            "  * " +
+            "FROM " +
+            "  (SELECT x, generate_series(1, x) FROM T1) tt " +
+            "WHERE x > 1"
```

Comment:
Should we also extend it to support cases where the query can be split? `WHERE x > 1 AND y > 10` (`y` being the generate_series result)

Could use `AndOperator.split` to do that.